package/android/src/main/java/com/mrousavy/camera/core/PreviewView.kt

```diff
@@ -8,22 +8,19 @@ import android.util.Size
 import android.view.Surface
 import android.view.SurfaceHolder
 import android.view.SurfaceView
+import com.mrousavy.camera.extensions.bigger
 import com.mrousavy.camera.extensions.getPreviewSize
+import com.mrousavy.camera.extensions.smaller
 import kotlin.math.roundToInt
 
 @SuppressLint("ViewConstructor")
 class PreviewView(context: Context,
-                  cameraManager: CameraManager,
-                  cameraId: String,
+                  private val targetSize: Size,
                   private val onSurfaceChanged: (surface: Surface?) -> Unit): SurfaceView(context) {
-  private val targetSize: Size
-  private val aspectRatio: Float
-    get() = targetSize.width.toFloat() / targetSize.height.toFloat()
+  private val aspectRatio: Double
+    get() = targetSize.smaller.toDouble() / targetSize.bigger
 
   init {
-    val characteristics = cameraManager.getCameraCharacteristics(cameraId)
-    targetSize = characteristics.getPreviewSize()
-
```

Comment:
Nice, I like that we can get rid of passing around cameraManager/cameraId here! :)